walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/resolvers/OpenIdURLRequestResolver.kt

```diff
@@ -32,7 +37,41 @@ internal class OpenIdURLRequestResolver(val libraryConfiguration: LibraryConfigu
             "Provided VerifiedIdRequestInput is not supported."
         )
         if (libraryConfiguration.isPreviewFeatureEnabled(PreviewFeatureFlags.FEATURE_FLAG_OPENID4VCI_ACCESS_TOKEN))
-            TODO("Not implemented")
+            return resolveOpenId4VCIRequest(verifiedIdRequestInput)
         return OpenIdResolver.getRequest(verifiedIdRequestInput.url.toString())
     }
+
+    private suspend fun resolveOpenId4VCIRequest(verifiedIdRequestInput: VerifiedIdRequestURL): Any {
+        val requestUri = getRequestUri(verifiedIdRequestInput.url)
+            ?: throw RequestURIMissingException("Request URI is not provided in ${verifiedIdRequestInput.url}.")
+        fetchOpenID4VCIRequest(requestUri)
+            .onSuccess { requestPayload ->
+                return try {
+                    JSONObject(requestPayload.decodeToString())
```

Comment:
nit: maybe create a variable to stringifiedRequestPayload or something to only decode into a string once?

Reply:
I will add a comment. That line is only to check if it is a valid json string. It is not assigned or used.

Reply:
but isnt it done again on the next line?

Reply:
First line checks if string is valid json and second line returns the decoded string. I see what you mean. Changed it.